pkg/build/apis/build/validation/validation.go

```diff
@@ -699,7 +699,13 @@ func ValidateStrategyEnv(vars []kapi.EnvVar, fldPath *field.Path) field.ErrorLis
 
 func validatePostCommit(spec buildapi.BuildPostCommitSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if spec.Script != "" && len(spec.Command) > 0 {
+	if len(spec.Command) == 0 && len(spec.Args) != 0 {
```

Comment:
we can't change validation without a migration.  if someone already has a buildconfig w/ invalid values, this will cause a mess for them.  (for example the build controller won't be able to update the build w/ the new build sequence number when they try to kick off the next build).

for now the important question is, what's the behavior if someone is in this state?